### PR TITLE
M058: OOI extern services integration to coi-services eoi

### PR DESCRIPTION
**Pull request by birdage**
M058:
Integration of externalization services to coi-services, requires update to pyon.yml for use pull request [here](https://github.com/ooici/ion-definitions/pull/525). Allows EOI services to piggy page on CRUD operations in dataset management service. To actually leverage the services it requires that the importerService be setup and running correctly, and that the eoi.meta.use_eoi_services be set in the pyon.yml file.

### CODE REVIEW

**Comment by birdage** [comment]
@jamie-cyber1 ready to merge, additional pull request to be satisfied first, located [here](https://github.com/ooici/ion-definitions/pull/525)

File: ion/services/dm/inventory/dataset_management_service.py [hunk 1]
```diff
@@ -27,6 +27,8 @@
 from coverage_model import NumexprFunction, PythonFunction, QuantityType, ParameterFunctionType
 from interface.objects import DataProcessDefinition, DataProcessTypeEnum, ParameterFunctionType as PFT
 
+from ion.services.eoi.tableLoader import resource_parser
```
Comment (mmeisinger) [inline ion/services/dm/inventory/dataset_management_service.py:30]: Please use PEP8 naming conventions. tableLoader is not a legal name. Use table_loader instead or one word.

File: ion/services/dm/inventory/dataset_management_service.py [hunk 2]
```diff
@@ -103,6 +127,12 @@ def update_dataset(self, dataset=None):
             raise BadRequest('%s: Dataset either not provided or malformed.' % self.logging_name)
         self.clients.resource_registry.update(dataset)
         #@todo: Check to make sure retval is boolean
+
+        #print "DM:update dataset:" + "dataset_id:" + dataset._id
+        if (self.geos_available):
```
Comment (mmeisinger) [inline ion/services/dm/inventory/dataset_management_service.py:132]: no need for parantheses

File: ion/services/eoi/tableLoader.py [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+import os
+from pyon.public import CFG
+
+DEBUG = False
+
+REAL = "real,"
+INT = "int,"
+TIMEDATE = "timestamp,"
+
+class resource_parser():
```
Comment (mmeisinger) [inline ion/services/eoi/tableLoader.py:25]: Class names use CamelCase, see PEP8 and must inherit from object)
class ResourceParser(object)

File: ion/services/eoi/tableLoader.py [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+        
+        try:
+            self.con = psycopg2.connect(database=self.DATABASE, user=self.DB_USER,password=self.DB_PASS)
+            self.cur = self.con.cursor()
+            #checks the connection
+            self.cur.execute('SELECT version()')
+            ver = self.cur.fetchone()
+            self.postgres_db_availabe = True
+            self.importer_service_available = self.checkForImporterService();
+            print ver
```
Comment (mmeisinger) [inline ion/services/eoi/tableLoader.py:60]: No prints in production code. Use log.debug

File: ion/services/eoi/tableLoader.py [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+            #SERVICE IS REALLY NOT AVAILABLE
+            print "service is really not available..."
+            return False
+
+    def close(self):
+        if self.con:
+            self.con.close()
+
+
+    def sendGeoNodeRequest(self,request,resource_id, prim_types=None):
```
Comment (mmeisinger) [inline ion/services/eoi/tableLoader.py:92]: illegal function name, see PEP8
use def send_geo_node_request
Also: spaces after commas

File: ion/services/eoi/tableLoader.py [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+        try:
+            
+            if prim_types is None:
+                r = requests.get(self.SERVER+'/service='+request+'&name='+resource_id+'&id='+resource_id)
+                self.processStatusCode(r.status_code) 
+            else:
+                r = requests.get(self.SERVER+'/service='+request+'&name='+resource_id+'&id='+resource_id+"&params="+str(prim_types))
+                self.processStatusCode(r.status_code) 
+                
+        except Exception, e:
```
Comment (mmeisinger) [inline ion/services/eoi/tableLoader.py:102]: use the newer code pattern with as
except Exception as e:

File: ion/services/eoi/tableLoader.py [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+            raise e
+
+    '''
```
Comment (mmeisinger) [inline ion/services/eoi/tableLoader.py:105]: For multi-line quotes, use double quotes:
"""
Comment
"""

**Comment by birdage** [comment]
@mmeisinger would the pull request be denied until these changes are met?

**Comment by birdage** [comment]
@jamie-cyber1 @oceanzus 
closed for now until above comments are met

**Comment by tampe** [comment]
Thanks Andy.  Please make sure to apply these standards throughout the whole set of code covered by this pull request and not just the ones identified in the comments from Michael.  Definitely should be no "print" statements in any service code - even if for debug.  Print statements should be migrated to use the logging mechanism we have with coi-services.

**Comment by birdage** [comment]
@tampe roger

**Pull request closed without merging**